pkg/apis/core/validation/validation.go

```diff
@@ -3083,8 +3083,32 @@ func validateContainersOnlyForPod(containers []core.Container, fldPath *field.Pa
 	return allErrs
 }
 
+// PodValidationOptions contains the different settings for pod validation
+type PodValidationOptions struct {
+	// Allow pod spec to have more than one huge page resource (with different sizes)
+	AllowMultipleHugePageResources bool
+}
+
+// ValidatePodSingleHugePageResources
+func ValidatePodSingleHugePageResources(pod *core.Pod, specPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	hugePageResources := sets.NewString()
+	for i := range pod.Spec.Containers {
+		resourceSet := toContainerResourcesSet(&pod.Spec.Containers[i])
+		for resourceStr := range resourceSet {
+			if v1helper.IsHugePageResourceName(v1.ResourceName(resourceStr)) {
+				hugePageResources.Insert(resourceStr)
+			}
+		}
+	}
+	if len(hugePageResources) > 1 {
+		allErrs = append(allErrs, field.Invalid(specPath, hugePageResources, "must use a single hugepage size in a pod spec"))
```

Comment:
suggest including the list of hugepages resources found in the error message

Reply:
They're already included I believe. Here is an error example:
```
[spec.containers[0].resources.limits: Required value: Limit must be set for non overcommitable resources spec: Invalid value: sets.String{"hugepages-1Gi":sets.Empty{}, "hugepages-2Mi":sets.Empty{}}: must use a single hugepage size in a pod spec]
```

Reply:
you're right, I missed that. pass in `hugePageResources.List()` to clean up that message so we don't log a map

Reply:
done. here is how it looks like now:
```
[spec.containers[0].resources.limits: Required value: Limit must be set for non overcommitable resources spec: Invalid value: []string{"hugepages-1Gi", "hugepages-2Mi"}: must use a single hugepage size in a pod spec]
```